discord/utils.py

```diff
@@ -334,6 +339,8 @@ def oauth_url(
         url += '&response_type=code&' + urlencode({'redirect_uri': redirect_uri})
     if disable_guild_select:
         url += '&disable_guild_select=true'
+    if state is not MISSING:
+        url += f'&state={state}'
```

Comment:
```suggestion
        encoded = urlencode({'state': state})
        url += f'&{encoded}'
```

Reply:
Ahh, I totally forgot about the encoding part